### PR TITLE
Upgrade to Solana 1.10.31

### PR DESCRIPTION
**Pull request by kskalski**

### CODE REVIEW

File: src/convert.rs [hunk 1]
```diff
@@ -669,6 +671,21 @@ impl TryFrom<tx_by_addr::TransactionError> for TransactionError {
                     ie,
                 ));
             }
+        } else if transaction_error.transaction_error == 30
```
Comment (lijunwangs) [inline src/convert.rs]: The code in validator which this file is copied from is:

        if let Some(transaction_details) = transaction_error.transaction_details {
            match transaction_error.transaction_error {
                30 => {
                    return Ok(TransactionError::DuplicateInstruction(
                        transaction_details.index as u8,
                    ));
                }
                31 => {
                    return Ok(TransactionError::InsufficientFundsForRent {
                        account_index: transaction_details.index as u8,
                    });
                }
                _ => {}
            }
        }

File: src/convert.rs [hunk 2]
```diff
@@ -798,6 +815,12 @@ impl From<TransactionError> for tx_by_addr::TransactionError {
                 TransactionError::WouldExceedAccountDataTotalLimit => {
                     tx_by_addr::TransactionErrorType::WouldExceedAccountDataTotalLimit
                 }
+                TransactionError::InsufficientFundsForRent { .. } => {
```
Comment (lijunwangs) [inline src/convert.rs]: Make it consistent with validator code:

                TransactionError::DuplicateInstruction(_) => {
                    tx_by_addr::TransactionErrorType::DuplicateInstruction
                }
                TransactionError::InsufficientFundsForRent { .. } => {
                    tx_by_addr::TransactionErrorType::InsufficientFundsForRent
                }

File: src/convert.rs [hunk 3]
```diff
@@ -967,6 +990,18 @@ impl From<TransactionError> for tx_by_addr::TransactionError {
                         },
                     })
                 }
+                TransactionError::InsufficientFundsForRent { account_index } => {
```
Comment (lijunwangs) [inline src/convert.rs:1002]: In v1.10.31 of validator:

            },
            transaction_details: match transaction_error {
                TransactionError::DuplicateInstruction(index) => {
                    Some(tx_by_addr::TransactionDetails {
                        index: index as u32,
                    })
                }
                TransactionError::InsufficientFundsForRent { account_index } => {
                    Some(tx_by_addr::TransactionDetails {
                        index: account_index as u32,
                    })
                }
                _ => None,
            },

**Review by lijunwangs — COMMENTED** [review]
Looks mostly good -- some nits which I would like to keep consistent with the validator code for ease of maintenance. Maybe we can consider remove it and import from validator directly in the future.

**Comment by kskalski** [comment]
Ah, I didn't realise this code is copied from solana/storage-proto

**Review by lijunwangs — APPROVED** [review]
Looks good

**Pull request merged**